### PR TITLE
vmm: Make gdb break/resuming more resilient

### PR DESCRIPTION
**Pull request by rbradford**
When starting the VM such that it is already on a breakpoint (via
stop_on_boot) when attached to gdb then start the vCPUs in a paused
state rather than starting the vCPUs later (upon resume).

Further, make the resumption/break of the VM more resilient by only
attempting to resume the vCPUs if were are already in a break point and
only attempting to pause/break if we were already running.

Fixes: #4354

Signed-off-by: Rob Bradford <robert.bradford@intel.com>

### CODE REVIEW

**Comment by rbradford** [comment]
This also addresses the `stepi` issue from from #4331

File: vmm/src/vm.rs [hunk 1]
```diff
@@ -2885,9 +2883,10 @@ impl Debuggable for Vm {
     }
 
     fn debug_pause(&mut self) -> std::result::Result<(), DebuggableError> {
-        if !self.cpu_manager.lock().unwrap().vcpus_paused() {
```
Comment (likebreath) [inline vmm/src/vm.rs:2888]: Nit: I believe `CpuManager::vcpus_paused()` can be dropped now.

**Review by likebreath — APPROVED** [review]
Took me a while to understand what behavior changes this PR introduce, but now the pause/resume logic with GDB looks much cleaner to me.

**Comment by michael2012z** [comment]
I tested the code with the `GDB on AArch64` PR https://github.com/cloud-hypervisor/cloud-hypervisor/pull/4355, both breakpoint and single-step work well.

**Review by michael2012z — APPROVED** [review]

**Review by sboeuf — APPROVED** [review]

**Pull request merged**